src/aatoolbox/datasources/usgs/usgs_ndvi.py

```diff
@@ -0,0 +1,815 @@
+            Drop modified column, used for merging data frames
+            together.
+
+        Returns
+        -------
+        pd.DataFrame
+            Processed data frame
+        """
+        df = pd.read_csv(filepath, parse_dates=["date", "modified"])
+        if drop_modified:
+            df.drop(["modified"], axis=1, inplace=True)
+        return df
+
+    def _get_raw_path(self, year: int, dekad: int, local: bool) -> Path:
+        """Get raw filepath.
+
+        Parameters
+        ----------
+        year : int
+            4-digit year
+        dekad : int
+            Dekad
+        local : bool
+            If True, returns filepath for local storage,
+            which includes full 4-digit year and _
+            separating with dekad. If False, filepath
+            corresponds to the zip file stored in the
+            USGS server.
+
+        Returns
+        -------
+        Path
+            Path to raw file
+        """
+        filename = self._get_raw_filename(year=year, dekad=dekad, local=local)
+        return self._raw_base_dir / f"{filename}.tif"
+
+    def _get_modified_time(self, year: int, dekad: int) -> datetime:
+        """Get modified time of raw file.
+
+        Used to determine when to re-process
+        or re-download raw raster files.
+
+        Parameters
+        ----------
+        year : int
+            4-digit year
+        dekad : int
+            Dekad
+
+        Returns
+        -------
+        datetime
+            Timestamp of when file was modified.
+        """
+        filepath = self._get_raw_path(year=year, dekad=dekad, local=True)
+        return datetime.fromtimestamp(filepath.stat().st_mtime)
+
+    def _get_processed_filename(self, feature_col: str, stat: str) -> str:
+        """Return processed filename.
+
+        Returns the processed filename. The suffix
+        of the filename is always the ``feature_col``
+        the statistics are aggregated to.
+
+        Returns
+        -------
+        str
+            Processed filename
+        """
+        base_file_name = self._get_processed_base_filename(
+            feature_col=feature_col
+        )
+        return f"{base_file_name}_{stat}.csv"
+
+    def _get_processed_base_filename(self, feature_col: str) -> str:
+        base_file_name = (
+            f"{self._country_config.iso3}"
+            f"_usgs_ndvi_{self._data_variable}"
+            f"_{feature_col}"
+        )
+        return base_file_name
+
+    def _get_processed_path(self, feature_col: str, stat: str) -> Path:
+        return self._processed_base_dir / self._get_processed_filename(
+            feature_col=feature_col, stat=stat
+        )
+
+    def _find_processed_files(self, feature_col: str) -> Generator:
+        base_file_name = self._get_processed_base_filename(
+            feature_col=feature_col
+        )
+        return self._processed_base_dir.glob(f"{base_file_name}_*.csv")
+
+    def _get_url(self, filename) -> str:
```

Comment:
I feel like this might be an important method to test -- if instead of mocking out `_USGSNdvi._download`, you mock out `urlopen` and `ZipFile` then it should be pretty straightforward I think....

Reply:
I think we can just test this method on its own, because we can't validate whether the URL is correct, so just validating the attributes and filename are passed correctly.